ckanext/relationships/helpers.py

```diff
@@ -118,6 +135,83 @@ def get_lineage_notes(type, object):
     return ''
 
 
-def get_relationship_types():
+def get_relationship_types(field=None):
     types = PackageRelationship.types
     return types
+
+
+def get_relationship_types_as_flat_list():
+    relationship_types = []
+
+    relationship_type_pairs = get_relationship_types()
+
+    for pair in relationship_type_pairs:
+        for value in pair:
+            if value:
+                relationship_types.append(value)
+
+    return relationship_types
+
+
+def quote_uri(uri):
+    from urllib.parse import quote
+    return quote(uri, safe='')
+
+
+def unquote_uri(uri):
+    from urllib.parse import unquote
+    return unquote(uri)
+
+
+# @TODO: Should this be moved into the `ckanext-qdes-schema` extension? YES
+def reconcile_package_relationships(context, pkg_id, related_resources):
```

Comment:
@MarkCalvert - I added this function to handle deleting only the relationships that are no longer in the `related_resources` field for the dataset.